### PR TITLE
feat: add "startEpoch" to round details

### PR DESCRIPTION
**Pull request by bajtos**
In order to map SPARK rounds to DRAND randomness values, we need to know the epoch (block number) when the SPARK round started.

This pull request modifies the round tracker to query the start epoch and store in the database, and the REST API to include this new field in the round details responses.

Links:
- https://github.com/filecoin-station/spark/issues/27
- https://www.notion.so/pl-strflt/SPARK-tasking-v2-604e26d57f6b4892946525bcb3a77104#8fe99341a9a447b98f198021d536cc05

### CODE REVIEW

File: lib/round-tracker.js [hunk 1]
```diff
@@ -225,3 +237,27 @@ async function defineTasksForRound (pgClient, sparkRoundNumber) {
     TASKS_PER_ROUND
   ])
 }
+
+/**
+ * @param {Awaited<ReturnType<import('./ie-contract.js').createMeridianContract>>} contract
+ * @param {bigint} roundIndex
+ * @returns {Promise<biging>} Filecoin Epoch (ETH block number) when the SPARK round started
```
Comment (juliangruber) [inline lib/round-tracker.js:244]:
```suggestion
 * @returns {Promise<bigint>} Filecoin Epoch (ETH block number) when the SPARK round started
```

File: lib/round-tracker.js [hunk 2]
```diff
@@ -225,3 +237,27 @@ async function defineTasksForRound (pgClient, sparkRoundNumber) {
...
+ */
+export async function getRoundStartEpoch (contract, roundIndex) {
+  assert.strictEqual(typeof roundIndex, 'bigint', `roundIndex must be a bigint, received: ${typeof roundIndex}`)
+
+  // Look at the last 1000 blocks to handle the case when we have an outage and
+  // the rounds are not advanced frequently enough.
+  const recentRoundStartEvents = (await contract.queryFilter('RoundStart', -1000))
+    .map(({ blockNumber, args }) => ({ blockNumber, roundIndex: args[0].toBigInt() }))
```
Comment (juliangruber) [inline lib/round-tracker.js:252]: It's unfortunate that the `contract.on(ev, fn)` doesn't expose the block number

File: migrations/049.do.round-start-epoch.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+ALTER TABLE spark_rounds
+ADD COLUMN start_epoch BIGINT;
+
+UPDATE spark_rounds SET start_epoch = 0;
+
+ALTER TABLE spark_rounds
+ALTER COLUMN start_epoch DROP NOT NULL;
```
Comment (juliangruber) [inline migrations/049.do.round-start-epoch.sql:7]:
```suggestion
ALTER COLUMN start_epoch SET NOT NULL;
```

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: lib/round-tracker.js (hunk 2 above)
Comment (bajtos) [reply lib/round-tracker.js:252]: Agreed 😢

File: migrations/049.do.round-start-epoch.sql (hunk 3 above)
Comment (bajtos) [reply migrations/049.do.round-start-epoch.sql]: Great catch!

File: lib/round-tracker.js (hunk 1 above)
Comment (bajtos) [reply lib/round-tracker.js]: It's time to setup TypeScript checks in this repo too 💪🏻

**Review by juliangruber — APPROVED** [review]

**Pull request merged**